dart/lib/src/sentry_options.dart

```diff
@@ -526,6 +530,13 @@ typedef BeforeBreadcrumbCallback = Breadcrumb? Function(
   Hint? hint,
 });
 
+/// This function is called right before a metric is about to be emitted.
+/// Can return true to emit the metric, or false to drop it.
+typedef BeforeMetricCallback = bool Function(
```

Comment:
I guess it's by design that we are not providing the `metric` to the callback, like `BeforeBreadcrumbCallback` provides the `breadcrumb` and drops it when `null` is returned? That way it would work the same as the other `Before` callbacks.

Reply:
Yep, this is how all the SDKs implement it.
I'd prefer to have the `BeforeMetricCallback` behave the same as the others, too.